sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -394,36 +453,131 @@ export class CryptographyClient {
-    } else {
-      if (!isLocallySupported(algorithm)) {
-        throw new Error(`Algorithm ${algorithm} is not supported for a local JsonWebKey.`);
-      }
-      return this.concreteClient.client.verifyData(
-        algorithm as LocalSupportedAlgorithmName,
-        data,
-        signature
-      );
+    this.ensureValid(await this.fetchKey(), KnownKeyOperations.Verify);
+    const { span, updatedOptions } = this.createSpan("encrypt", options);
+
+    try {
+      const provider = await this.getProvider(KnownKeyOperations.Verify, algorithm);
+      const digest = await createHash(algorithm, data);
+      return await provider.verify(algorithm, digest, signature, updatedOptions);
+    } finally {
+      span.end();
     }
   }
 
   /**
-   * The base URL to the vault.
-   * Will be empty if called on a local key.
+   * @internal
+   * Retrieves the {@link JsonWebKey} from the Key Vault.
+   *
+   * Example usage:
+   * ```ts
+   * let client = new CryptographyClient(keyVaultKey, credentials);
+   * let result = await client.getKey();
+   * ```
+   * @param options - Options for retrieving key.
    */
-  get vaultUrl(): string {
-    return this.concreteClient.client.vaultUrl;
+  private async getKeyMaterial(): Promise<JsonWebKey> {
+    const key = await this.fetchKey();
+
+    switch (key.kind) {
+      case "JsonWebKey":
+        return key.value;
+      case "KeyVaultKey":
+        return key.value.key!;
+      default:
+        throw new Error("Failed to exchange Key ID for an actual KeyVault Key.");
+    }
   }
 
   /**
-   * The ID of the key used to perform cryptographic operations for the client.
+   * Returns the underlying key used for cryptographic operations.
+   * If needed, fetches the key from KeyVault and exchanges the ID for the actual key.
+   * @param options - The additional options.
    */
-  get keyId(): string | undefined {
-    return this.concreteClient.client.keyId;
+  private async fetchKey(options: GetKeyOptions = {}): Promise<CryptographyClientKey> {
+    if (this.key.kind === "identifier") {
+      // Exchange the identifier with the actual key when needed
+      const key = await this.remoteProvider!.getKey(options);
+      this.key = { kind: "KeyVaultKey", value: key };
+    }
+    return this.key;
+  }
+
+  private providers?: CryptographyProvider[];
+  /**
+   * Gets the provider that support this algorithm and operation.
+   * The available providers are ordered by priority such that the first provider that supports this
+   * operation is the one we should use.
+   * @param operation - The {@link KeyOperation}.
+   * @param algorithm - The algorithm to use.
+   */
+  private async getProvider(
+    operation: KeyOperation,
+    algorithm: string
+  ): Promise<CryptographyProvider> {
+    if (!this.providers) {
+      const keyMaterial = await this.getKeyMaterial();
+      this.providers = [];
+
+      // Local crypto providers in node only.
+      if (isNode) {
```

Comment:
@xirzec  I'm not familiar with rollup at all, but would this be sufficient to exclude RsaCryptographyProvider and all its imports from the generated bundle?

Reply:
We generally like to add a different file for the browser implementation. Ping me and I'll show you. It ends up being way cleaner.

Reply:
Updated. Keeping this conversation open in case I did it wrong 😄